service/controller/key/cluster.go

```diff
@@ -114,6 +116,38 @@ func IsChinaRegion(awsRegion string) bool {
 	return strings.HasPrefix(awsRegion, "cn-")
 }
 
+func IsNewCluster(cluster infrastructurev1alpha2.AWSCluster) bool {
+	// If  condition  list is empty then this is new cluster.
+	if len(cluster.Status.Cluster.Conditions) == 0 {
+		return true
+	}
+	// If versions list is empty then this is a new cluster.
+	if len(cluster.Status.Cluster.Versions) == 0 {
+		return true
+	}
+	// Check if there is transition state Updated or Updating,
+	// this indicates cluster is not new but updated from other version.
+	for _, condition := range cluster.Status.Cluster.Conditions {
+		if condition.Condition == "Updating" {
+			return false
+		}
+		if condition.Condition == "Updated" {
+			return false
+		}
+	}
+
+	// Check if there is transition state from other version,
+	// this indicates cluster is not new but updated from other version.
+	for _, version := range cluster.Status.Cluster.Versions {
+		if version.Version != project.Version() {
+			return false
+		}
+	}
```

Comment:
We have [HasVersion](https://godoc.org/github.com/giantswarm/apiextensions/pkg/apis/infrastructure/v1alpha2#CommonClusterStatus.HasVersion).

Reply:
I think that does not work here, as we need to check if there is NOT some other version rather than check if there is some specific version

Reply:
Then you can use [LatestVersion](https://godoc.org/github.com/giantswarm/apiextensions/pkg/apis/infrastructure/v1alpha2#CommonClusterStatus.LatestVersion). If it is empty the cluster is new. If it is the current version the cluster is new. Any other output means the cluster is updated.

Reply:
But ok, your call. Do what you think is right. 🤗